### PR TITLE
Add date time conversion & formatting functions

### PR DESCRIPTION
**Pull request by uklotzde**
Very simple functions with debug assertions to detect any unexpected values early.

As discussed in #2698.

### CODE REVIEW

File: src/library/browse/browsethread.cpp [hunk 1]
```diff
@@ -248,16 +249,20 @@ void BrowseThread::populateModel() {
             item->setData(location, Qt::UserRole);
             row_data.insert(COLUMN_NATIVELOCATION, item);
 
-            QDateTime modifiedTime = pTrack->getFileInfo().fileLastModified().toLocalTime();
-            item = new QStandardItem(modifiedTime.toString(Qt::DefaultLocaleShortDate));
+            const auto fileLastModified =
+                    pTrack->getFileInfo().fileLastModified();
+            item = new QStandardItem(
+                    mixxx::displayDateTime(fileLastModified));
```
Comment (Holzhaus) [inline src/library/browse/browsethread.cpp:255]:
```suggestion
                    mixxx::displayDateTime(fileLastModified.toLocalTime()));
```
Or is this not necessary?

Comment (uklotzde) [reply src/library/browse/browsethread.cpp:255]: I checked it. No, not necessary. Qt::DefaultLocaleShortDate already performs the conversion into the local time zone. I will add a comment for the function.

Comment (uklotzde) [reply src/library/browse/browsethread.cpp:255]: I will rename the function: displayLocalDateTime()

**Comment by ronso0** [comment]
Works fine, tested with Tracks, Browse and DlgTrackInfo in #2698

File: src/util/datetime.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace mixxx {
+
+/// Common utility functions for safely converting and consistently
+/// displaying date time values.
+
+/// Obtain the local date time from an UTC date time.
+inline QDateTime localDateTimeFromUtc(
+        const QDateTime& dt) {
+    DEBUG_ASSERT(dt.timeSpec() == Qt::UTC);
```
Comment (uklotzde) [inline src/util/datetime.h:16]: The original code called setTimeSpec(Qt::UTC) without any explanation why. I added this debug assertion to check that it is indeed unnecessary. 

Wouldn't do any harm other than displaying the time with an offset so I guess this is acceptable.

Comment (uklotzde) [reply src/util/datetime.h]: The debug assertion was triggered so I reverted back to the original code.

**Review by ronso0 — APPROVED** [review]
LGTM!

**Pull request merged**